### PR TITLE
Rc correct grade sheet endpoints

### PR DESCRIPTION
**Pull request by robert-czwartosz**
Corrected grade sheet endpoints:
GET /api/grade/sheets/:id --> get grade sheet
GET /api/grade/sheets/:id/reviewers/:mentorId/grades --> get grades for reviewer from given sheet
GET /api/participants/:id/grade/sheets --> get grade sheets for given participant
GET /api/mentors/id/grade/sheets --> get grade sheets for given mentor
GET /api/reviewers/:id/grade/sheets --> get grade sheets for given reviewer

POST /api/grade/sheets/:id/add/reviewer/:mentorId --> add mentor reviewer to sheet
POST /api/grade/sheets/:id/add/participant/:participantId --> add participant to sheet

PUT /api/grade/sheets/:id/reviewers --> set reviewers array
    body: { reviewers: ObjectId[]; }
PUT /api/grade/sheets/:id/participants --> set participants array
    body: { participants: Participant[] }
    Participant: { participantID: ObjectID; enggagement?: number; role?: string; rolePoints?: number; }

PATCH /api/grade/sheets/:id/participants --> update participants
    body: { participants: Participant[] }
PATCH /api/grade/sheets/:id/mentor/grades --> set mentor grades
    body: { [gradeName: string]: number; }
PATCH /api/grade/sheets/:id/reviewers/:mentorId/grades --> set mentor reviewer grades
    body: { [gradeName: string]: number; }

DELETE /api/grade/sheets/:id/participants/:participantId --> delete participant
DELETE /api/grade/sheets/:id --> delete grade sheet

### CODE REVIEW

File: Api/Src/Models/GradeSheet.ts [hunk 1]
```diff
@@ -28,6 +31,11 @@ const GradeSheetSchema = new mongoose.Schema({
         ref: 'Project',
```
Comment (kjreczek) [inline Api/Src/Models/GradeSheet.ts:31]: There should be ref to TeamProject, not Project

Comment (robert-czwartosz) [reply Api/Src/Models/GradeSheet.ts]: Thanks, corrected 👍

File: Api/Src/Services/GradeSheetService.ts [hunk 2]
```diff
@@ -46,15 +59,75 @@ export default class GradeSheetService {
...
+        sheet.reviewers = sheet.reviewers
+            .filter(rev => mentorIds.some(id => `${rev}` === `${id}`));
+        sheet.mentorReviewerGrades = sheet.mentorReviewerGrades
+            .filter(grade => mentorIds.some(id => `${grade.mentorID}` === `${id}`));
+        for (let i in mentorIds) {
+            if (sheet.reviewers.some(rev => `${mentorIds[i]}` === `${rev}`)) continue;
+            sheet.reviewers.push(mentorIds[i]);
+            sheet.mentorReviewerGrades.push({mentorID: mentorIds[i], grades: {}});
+        }
+        return await this.repository.save(sheet);
```
Comment (fszwedo) [inline Api/Src/Services/GradeSheetService.ts:72]: it is possible to set mentor reviewer to non-existing user id
I think we can live with that though as this is manageable on frontend

**Review by fszwedo — APPROVED** [review]
I don't understand how to get and patch mentor grades I think, every time I do it I get the info that grade was saved and then I get an empty object when I want to retrieve the grades 

@robert-czwartosz could you please let me know how to test 
GET /api/grade/sheets/:id/reviewers/:mentorId/grades --> get grades for reviewer from given sheet
PATCH /api/grade/sheets/:id/reviewers/:mentorId/grades --> set mentor reviewer grades

apart from that all seems to be working fine, as the issue with my review is most probably on my side I'm approving the request not to block it (I'd like to test the missing parts though)

Codewise I only see that we could validate if reviewers/participants exist before adding them to the grade sheet but this is something we can do on frontend as well

**Review by kjreczek — APPROVED** [review]

**Pull request merged**